File: back_end/src/main/java/com/example/back_end/entity/Manufacturer.java
package com.example.back_end.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "manufacturer")
public class Manufacturer extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "name", length = Integer.MAX_VALUE)
    private String name;

    @Column(name = "description", length = Integer.MAX_VALUE)
    private String description;

    @Column(name = "picture_id")
    private Integer pictureId;

    @Column(name = "page_size")
    private Integer pageSize;

    @Column(name = "price_range_filtering")
    private Boolean priceRangeFiltering;

    @Column(name = "published")
    private Boolean published;

    @Column(name = "deleted")
    private Boolean deleted;

    @Column(name = "display_order")
    private Integer displayOrder;

}

File: front_end/src/model/Manufacturer.ts
export type ManufacturerResponse ={
    id:  number,
    name: string,
    description: string,
    pictureId: number,
    pageSize: number
    published: boolean
    deleted: boolean
    displayOrder: number
    priceRangeFiltering: boolean
}
export type ManufacturerRequest={
    id?: number,
    name: string,
    description: string,
    pictureId: number | null,
    pageSize: number,
    published: boolean,
    displayOrder: number,
    priceRangeFiltering: boolean
}